torch/profiler/strobelight_cli_function_profiler.py

```diff
@@ -0,0 +1,311 @@
+        sample_each=1e7,  # sample each sample_each cycles.
+        run_user_name="pytorch-strobelight-ondemand",
+        timeout_wait_for_running_sec=60,
+        timeout_wait_for_finished_sec=60,
+        recorded_env_variables=None,
+        sample_tags=None,
+        stack_max_len=127,
+        async_stack_max_len=127,
+    ):
+        self.stop_at_error = stop_at_error
+        self.max_profile_duration_sec = max_profile_duration_sec
+        self.sample_each = sample_each
+        self.run_user_name = run_user_name
+        self.timeout_wait_for_running_sec = timeout_wait_for_running_sec
+        self.timeout_wait_for_finished_sec = timeout_wait_for_finished_sec
+        # Results of the most recent run.
+        # Tracks the strobelight run id of the most recent run
+        self.current_run_id = None
+        self.sample_tags = sample_tags
+
+    def _run_async(self):
+        processId = os.getpid()
+        namespace = pid_namespace(processId)
+        command = [
+            "strobeclient",
+            "run",
+            "--profiler",
+            "pyperf",
+            "--event",
+            "cycles",
+            "--async",
+            "--sample-interval",
+            str(int(self.sample_each)),
+            "--duration-ms",
+            str(int(self.max_profile_duration_sec * 1000)),
+            "--pid",
+            f"{namespace}:{processId}",
+        ]
+
+        if self.sample_tags:
+            command.append("--sample-tags")
+            add_comma = False
+            for item in self.sample_tags:
+                if add_comma:
+                    command.append(",")
+                command.append(item)
+                add_comma = True
+
+        logger.debug(f"running command:{command_to_string(command)}")
+
+        result = subprocess.run(command, capture_output=True)
+
+        if result.returncode != 0:
+            raise StrobelightCLIProfilerError(
+                "failed to start strobelight profiling, error in run_async"
+            )
+
+        for line in result.stderr.split(b"\n"):
+            logger.debug(line)
+            match = re.search(rb"INFO Run Id:.*", line)
+            if match:
+                self.current_run_id = int(
+                    re.search(rb"-?\d+$", match.group(0)).group(0)
+                )
+
+    def _wait_for_running(self, counter=0):
+        if counter > 20:
+            raise StrobelightCLIProfilerError(
+                "wait_for_running called more than 20 times"
+            )
+
+        command = ["strobeclient", "getRunStatus", "--run-id", str(self.current_run_id)]
+
+        logger.debug(f"running command:{command_to_string(command)}")
+
+        result = subprocess.run(command, capture_output=True)
+
+        if result.returncode != 0:
+            raise StrobelightCLIProfilerError(
+                "failed to start strobelight profiling, error in wait_for_running"
+            )
+
+        for line in result.stderr.split(b"\n"):
```

Comment:
Same deal as above - no need to split and search by lines - just search in the whole unsplit stderr.